api/cloud/oc_cloud_manager.c

```diff
@@ -458,8 +494,10 @@ oc_cloud_refresh_token_handler(oc_client_response_t *data)
   }
   free_api_param(p);
 
-  ctx->store.status &=
-    ~(OC_CLOUD_FAILURE | OC_CLOUD_TOKEN_EXPIRY | OC_CLOUD_REFRESHED_TOKEN);
+  ctx->store.status &= ~(OC_CLOUD_FAILURE | OC_CLOUD_TOKEN_EXPIRY | OC_CLOUD_REFRESHED_TOKEN);
+  if (data->code == OC_STATUS_UNAUTHORIZED) {
+    ctx->store.status &= ~(OC_CLOUD_REGISTERED | OC_CLOUD_LOGGED_IN);
+  }
```

Comment:
this part I think below to `_refresh_token_handler`

Reply:
something similar i did here https://github.com/iotivity/iotivity-lite/blob/master/api/cloud/oc_cloud_manager.c#L445

Reply:
I see, the master code has some equivalent updates to what I made here to jakarta.

Reply:
Pls merge master to you PR and move changes there.

Reply:
done, though there are plenty of commits there, but since they are in master already it shouldn't matter